kasa/modules/rulemodule.py

```diff
@@ -26,55 +34,114 @@ class TimeOption(Enum):
     AtSunset = 2
 
 
-class Rule(BaseModel):
+class BulbModeOption(Enum):
+    """Bulb mode."""
+
+    Customize = "customize_preset"
+    Last = "last_status"
+
+
+class BaseRule(BaseModel):
     """Representation of a rule."""
 
-    id: str
+    # not used when adding a rule
+    id: Optional[str]
     name: str
-    enable: bool
-    wday: List[int]
-    repeat: bool
+    enable: EnabledOption
+
+    class Config:
+        """Rule Config."""
+
+        validate_assignment = True
+        allow_population_by_field_name = True
+
+
+class CountdownRule(BaseRule):
+    """Representation of a countdown rule."""
+
+    delay: int
+    act: EnabledOption = Field(alias="action")
+
+
+class ScheduleRule(BaseRule):
+    """Representation of a schedule rule."""
+
+    wday: List[int] = Field(alias="days")
+    repeat: EnabledOption
 
     # start action
-    sact: Optional[Action]
-    stime_opt: TimeOption
-    smin: int
+    sact: Optional[Action] = Field(alias="start_action")
```

Comment:
```suggestion
    start_action: Optional[Action] = Field(alias="sact")
```
This is wrong way around, no? Basically, the alias allows using a nicer name for the field name itself.